mcs/class/System/Mono.UnityTls/UnityTlsContext.cs

```diff
@@ -183,19 +183,19 @@ public override (int ret, bool wantMore) Read (byte[] buffer, int offset, int co
 			if (lastException != null)
 				throw lastException;
 
-			if (errorState.code == UnityTls.unitytls_error_code.UNITYTLS_USER_WOULD_BLOCK)
-				wouldBlock = true;
+			if (errorState.code == UnityTls.unitytls_error_code.UNITYTLS_USER_WOULD_BLOCK || numBytesRead < count) // In contrast to some other APIs (like Apple security) WOULD_BLOCK is not set if we did a partial read
+				wantMore = true;
 			else if (errorState.code == UnityTls.unitytls_error_code.UNITYTLS_STREAM_CLOSED)
 				return (0, false);	// According to Apple and Btls implementation this is how we should handle gracefully closed connections.
-			else
+			elsegit 
```

Comment:
fancy else statement :)

Reply:
oh my. gosh, thanks! must have had wrong window focus when comiting the thing. Woops.